pkg/operator/controller/canary/daemonset.go

```diff
@@ -131,6 +131,11 @@ func canaryDaemonSetChanged(current, expected *appsv1.DaemonSet) (bool, *appsv1.
 		changed = true
 	}
 
+	if !cmp.Equal(current.Spec.Template.Spec.PriorityClassName, expected.Spec.Template.Spec.PriorityClassName, cmpopts.EquateEmpty()) {
```

Comment:
Any reason to use `cmp.Equal` instead of `==`?

Reply:
nope. Fixed.